ComponentKit/Utilities/CKComponentTriggerInternal.h

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <memory>
+
+#import <ComponentKit/CKComponentScope.h>
+#import <ComponentKit/CKComponentAction.h>
+
+/** An implementation detail class. Should not be used directly. */
+class CKComponentTriggerTargetBase {
+protected:
+  BOOL _resolved;
+  __weak CKComponentScopeHandle *_scopeHandle;
+  SEL _selector;
+
+  NSInvocation *invocation(CKComponent *sender) const;
+
+  void resolve(void);
+
+  CKComponentTriggerTargetBase();
+  ~CKComponentTriggerTargetBase();
+  void resolve(const CKComponentScope &scope, SEL selector);
+  explicit operator bool() const;
+
+  friend class CKComponentTriggerBase;
+};
+
+template <typename... T>
+class CKTypedComponentTrigger;
+
+template <typename... T>
+class CKTypedComponentTriggerTarget : public CKComponentTriggerTargetBase {
+  static_assert(std::is_same<
```

Comment:
Yeah, I kind of like this just using the same as actions, and for actions we have specified that you may specify as many parameters as you choose.